shared/AppInsightsCore/src/JavaScriptSDK/StatsBeat.ts

```diff
@@ -0,0 +1,216 @@
+
+          
+
+            function _sendStatsbeats() {
+                // Add extra properties
+                let networkProperties = {
+                    "cikey": _cikey,
+                    "runtimeVersion": _runTimeVersion,
+                    "language": _language,
+                    "version": _sdkVersion,
+                    "os": _os
+                }
+                if (objKeys(_statsbeatMetrics)) {
+                    let statsbeat: ITelemetryItem = {
+                        iKey: INSTRUMENTATION_KEY,
+                        name: NETWORK,
+                        baseData: {
+                            name: NETWORK,
+                            average: 0,
+                            properties: {"host": _networkCounter.host, ..._statsbeatMetrics.properties, ...networkProperties}
+                        }
+                        // baseType: Metric.dataType
+                    };
+                    _channel.processTelemetry(statsbeat);
+                }
+                _statsbeatMetrics = {};
+                _channel.flush(true);
+            }
+
+            function _trackRequestDuration() {
+                var currentCounter = _networkCounter;
+                currentCounter.time = utcNow();
+                var intervalRequests = (currentCounter.totalRequestCount - currentCounter.lastRequestCount) || 0;
+                var elapsedMs = currentCounter.time - currentCounter.lastTime;
+                var averageRequestExecutionTime = ((currentCounter.intervalRequestExecutionTime - currentCounter.lastIntervalRequestExecutionTime) / intervalRequests) || 0;
+                currentCounter.lastIntervalRequestExecutionTime = currentCounter.intervalRequestExecutionTime; // reset
+                if (elapsedMs > 0 && intervalRequests > 0) {
+                    _statsbeatMetrics.properties = _statsbeatMetrics.properties || {};
+                    _statsbeatMetrics.properties[StatsbeatCounter.REQUEST_DURATION] = averageRequestExecutionTime;
+                }
+                // Set last counters
+                currentCounter.lastRequestCount = currentCounter.totalRequestCount;
+                currentCounter.lastTime = currentCounter.time;
+            }
+
+            function _trackRequestsCount() {
+                var currentCounter = _networkCounter;
+                if (currentCounter.totalSuccesfulRequestCount > 0) {
+                    _statsbeatMetrics.properties = _statsbeatMetrics.properties || {};
+                    _statsbeatMetrics.properties[StatsbeatCounter.REQUEST_SUCCESS] = currentCounter.totalSuccesfulRequestCount;
+                    currentCounter.totalSuccesfulRequestCount = 0; //Reset
+                }
+                if (currentCounter.totalFailedRequestCount > 0) {
+                    _statsbeatMetrics.properties = _statsbeatMetrics.properties || {};
+                    _statsbeatMetrics.properties[StatsbeatCounter.REQUEST_FAILURE] = currentCounter.totalFailedRequestCount;
```

Comment:
It looks like as of this PR all failures are added to a generic "failure" counter. I don't see any accounting for specific status codes as separate counters or including status code as a custom dimension. Are you planning to add support for that attribute later?